fastlane/lanes/release.rb

```diff
@@ -88,14 +132,54 @@
   #####################################################################################
   desc 'Updates a release branch for a new beta release'
   lane :new_beta_release do |options|
-    android_betabuild_prechecks(base_version: options[:base_version], skip_confirm: options[:skip_confirm])
+    # Checkout default branch and update
+    Fastlane::Helper::GitHelper.checkout_and_pull(DEFAULT_BRANCH)
+
+    # Check local repo status
+    ensure_git_status_clean
+
+    # Check versions
+    message = <<-MESSAGE
+
+      Current beta version: #{current_beta_version}
+      New beta version: #{next_beta_version}
+
+      Current build code: #{current_build_code}
+      New build code: #{next_build_code}
+
+    MESSAGE
+
+    # Check branch
+    unless !options[:base_version].nil? || Fastlane::Helper::GitHelper.checkout_and_pull(release: current_release_version)
+      UI.user_error!("Release branch for version #{current_release_version} doesn't exist.")
+    end
+
+    # Check user override
+    override_default_release_branch(options[:base_version]) unless options[:base_version].nil?
+
+    # Verify
+    if options[:skip_confirm]
+      UI.message(message)
+    else
+      UI.user_error!('Aborted by user request') unless UI.confirm("#{message}Do you want to continue?")
+    end
+
     update_frozen_strings_for_translation
-    download_translations()
-    android_bump_version_beta()
+    download_translations
+
+    # Bump the release version and build code
+    UI.message 'Bumping beta version and build code...'
+    ensure_git_branch(branch: '^release/') # Match branch names that begin with `release/`
```

Comment:
Should we do this after `Fastlane::Helper::GitHelper.checkout_and_pull(release: current_release_version)`? If we are going to do this verification, we might as well do it before updating frozen strings and downloading translations.

Reply:
I decided to remove this check because we're already checking that the current version's release branch is being checked out here: 

```
unless Fastlane::Helper::GitHelper.checkout_and_pull(release: current_release_version)
      UI.user_error!("Release branch for version #{current_release_version} doesn't exist.")
    end
```

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-Android/commit/08c57ebfd46cef2dc7a0f4c8943fc58abc5f0d86